uniffi_bindgen/src/bindings/python/gen_python/visit_mut.rs

```diff
@@ -0,0 +1,569 @@
+            }
+            UniffiTrait::Hash { hash } => {
+                hash.name = "__hash__".to_string();
+            }
+        }
+        Ok(())
+    }
+
+    fn visit_argument(&mut self, arg: &mut Argument) -> Result<()> {
+        arg.name = self.var_name(&arg.name);
+        Ok(())
+    }
+
+    fn visit_constructor(&mut self, cons: &mut Constructor) -> Result<()> {
+        cons.name = if cons.is_primary_constructor() && !cons.is_async() {
+            "__init__".to_string()
+        } else {
+            self.fn_name(&cons.name)
+        };
+        cons.docstring = self.format_docstring(cons.docstring.take());
+        if cons.is_async() {
+            self.add_module_import("asyncio");
+            self.runtimes.async_ = true;
+        }
+        Ok(())
+    }
+
+    fn visit_function(&mut self, func: &mut Function) -> Result<()> {
+        func.name = self.fn_name(&func.name);
+        func.docstring = self.format_docstring(func.docstring.take());
+        if func.is_async() {
+            self.add_module_import("asyncio");
+            self.runtimes.async_ = true;
+        }
+        self.exports.push(func.name.clone());
+        Ok(())
+    }
+
+    fn visit_type(&mut self, ty: &mut Type) -> Result<()> {
+        self.add_imports_for_type(ty);
+        let ffi_converter_name = format!("_UniffiConverter{}", ty.canonical_name());
+        match &ty.kind {
+            TypeKind::External { namespace, .. } => {
+                let mod_name = self.config.module_for_namespace(namespace);
+                ty.lang_data.insert(
+                    "ffi_converter_name",
+                    format!("{mod_name}.{ffi_converter_name}"),
+                );
+            }
+            _ => ty
+                .lang_data
+                .insert("ffi_converter_name", ffi_converter_name),
+        }
+        ty.lang_data.insert("type_name", self.type_name(ty));
+        Ok(())
+    }
+
+    fn visit_return_type(&mut self, return_type: &mut ReturnType) -> Result<()> {
+        let ffi_default = match &return_type.ty {
+            Some(t) => match &t.ffi_type.kind {
+                FfiTypeKind::UInt8
+                | FfiTypeKind::Int8
+                | FfiTypeKind::UInt16
+                | FfiTypeKind::Int16
+                | FfiTypeKind::UInt32
+                | FfiTypeKind::Int32
+                | FfiTypeKind::UInt64
+                | FfiTypeKind::Int64
+                | FfiTypeKind::Handle => "0".to_string(),
+                FfiTypeKind::Float32 | FfiTypeKind::Float64 => "0.0".to_string(),
+                FfiTypeKind::RustArcPtr(_) => "ctypes.c_void_p()".to_string(),
+                FfiTypeKind::RustBuffer(meta) => match meta {
+                    None => "_UniffiRustBuffer.default()".to_string(),
+                    Some(meta) => {
+                        let module_name = self.config.module_for_namespace(&meta.namespace);
+                        format!("{module_name}._UniffiRustBuffer.default()")
+                    }
```

Comment:
This is another thing I like about the new system, we always have access to the `Config` which makes referencing the external rust buffer more straightforward.

Before we didn't want to do this because it means that `ffi_type_name()` and any other function that depended on it would need to input the config, which was deemed too annoying.